mifospay/src/main/java/org/mifospay/bank/setupUpi/screens/DebitCardScreen.kt

```diff
@@ -0,0 +1,134 @@
+) {
+
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(dimensionResource(id = R.dimen.value_20dp)),
+        elevation = 1.dp
+    ) {
+        Column(
+            modifier = Modifier.padding(
+                top = dimensionResource(id = R.dimen.value_15dp),
+                bottom = dimensionResource(id = R.dimen.value_15dp),
+                start = dimensionResource(id = R.dimen.value_10dp),
+                end = dimensionResource(id = R.dimen.value_10dp)
+            )
+        ) {
+            VerifyStepHeader("Debit Card Details ", verificationStatus)
+            if (isContentVisible) {
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(start = 12.dp)
+                ) {
+                    //handle debit card ui state
+                    Box(
+                        modifier = Modifier.fillMaxWidth()
+                    ) {
+                        DebitCardScreenContents()
+                        when (debitCardUiState) {
+                            is DebitCardUiState.Initials -> {
+
+                            }
+                            is DebitCardUiState.Verifying -> {
+                                MifosLoadingWheel(
+                                    modifier = Modifier
+                                        .wrapContentSize()
+                                        .align(Alignment.Center),
+                                    contentDesc = "Verifying Debit Card"
+                                )
+                            }
+                            is DebitCardUiState.Verified -> {
+                                onDebitCardVerified((debitCardUiState).otp)
+                            }
+                            is DebitCardUiState.VerificationFailed -> {
+                                onDebitCardVerificationFailed((debitCardUiState).errorMessage)
+                            }
+                            else -> {}
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Preview
+@Composable
+fun DebitCardScreenInitialsPreview() {
+    MifosTheme {
+        DebitCardScreenWithHeaderAndContent(debitCardUiState = DebitCardUiState.Initials,
+            onDebitCardVerified = {},
+            onDebitCardVerificationFailed = {})
+    }
+}
+
+@Preview
+@Composable
+fun DebitCardScreenLoadingPreview() {
+    MifosTheme {
+        DebitCardScreenWithHeaderAndContent(debitCardUiState = DebitCardUiState.Verifying,
+            onDebitCardVerified = {},
+            onDebitCardVerificationFailed = {})
+    }
+}
+
+@Preview
+@Composable
+fun DebitCardScreenVerificationFailedPreview() {
+    MifosTheme {
+        DebitCardScreenWithHeaderAndContent(debitCardUiState = DebitCardUiState.VerificationFailed("Error Message"),
+            onDebitCardVerified = {},
+            onDebitCardVerificationFailed = {})
+    }
+}
```

Comment:
use preview parameter to generate previews of differenet UI States. You can Refer NotificationScreen for this